src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java

```diff
@@ -88,6 +99,67 @@ public TarantoolTemplate(
         );
     }
 
+    private <T> CallResultMapper<T, SingleValueCallResult<T>>
+    getDefaultWithTupleResultMapper(Class<T> entityClass, Optional<TarantoolSpaceMetadata> spaceMetadata) {
+        // blocked because it's not deepcopy
+        // MessagePackMapper defaultConverterWithTupleConverter = tarantoolClient.getConfig().getMessagePackMapper().copy();
+        MessagePackMapper defaultConverterWithTupleConverter = DefaultMessagePackMapperFactory.getInstance().defaultComplexTypesMapper();
+        defaultConverterWithTupleConverter.registerValueConverter(
+                ArrayValue.class,
+                TarantoolTuple.class,
+                new TarantoolAutoResultConverter<>(
+                        new DefaultTarantoolTupleValueConverter(defaultConverterWithTupleConverter,
+                                                                spaceMetadata.orElse(null)),
+                        spaceMetadata.orElse(null)));
+
+        CallResultMapper<T, SingleValueCallResult<T>> resultMapper =
+                ((DefaultResultMapperFactoryFactory) tarantoolClient.getResultMapperFactoryFactory())
+                .getDefaultSingleValueMapper(defaultConverterWithTupleConverter, entityClass);
```

Comment:
This also looks bad, getDefaultSingleValueMapper can be obtained **only** from the DefaultResultMapperFactoryFactory. Note, without half a bottle you won't understand these identical SingleValueResultMappers ☹️

Reply:
Too much repetition here, a helper function/factory or a new superficial mapper class may improve this a bit.

Reply:
I didn’t quite understand what kind of repetition you are talking about here, that we create a new converter with each request?

Reply:
I would like to inherit the `DefaultResultMapperFactoryFactory` factory, but it is final in cartridge-java